src/models/artifact.ts

```diff
@@ -13,4 +15,11 @@ export type Artifact = {
   key: string,
   type: ArtifactType,
   isLatest: boolean,
-}
\ No newline at end of file
+}
+
+export const artifactTypeIconMap = {
+  markdown: 'ArtifactMarkdown',
+  table: 'ArtifactTable',
+  result: 'ArtifactResult',
+  unknown: 'Artifact',
+} as Record<ArtifactType, IconName>
```

Comment:
This will give you better type safety
```suggestion
} as const satisfies Record<ArtifactType, IconName>
```